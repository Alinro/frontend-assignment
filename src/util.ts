export const getFormData = (form: HTMLFormElement): string => {
    const formData = new FormData(form);
    const result: Record<string, FormDataEntryValue> = {};

    for (const [key, value] of formData.entries()) {
        result[key] = value;
    }

    return JSON.stringify(result);
}

export const showSuccessMessage = () => {
    const paragraph = document.createElement('p');
    paragraph.innerHTML =
        'Your payment information was successfully submitted. Please check the data in the devtool console';

    document.querySelector('#success')?.appendChild(paragraph);
}