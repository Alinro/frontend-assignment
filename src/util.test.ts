import { describe, expect, it } from 'vitest'
import { getFormData } from './util';

/**
 * @vitest-environment jsdom
 */
describe('test getFormData', () => {
    it('returns the correct value according to the snapshot, when the form has an input', () => {
        const input = document.createElement('input');
        input.value = 'inputValue';
        input.name = 'inputName';

        const form = document.createElement('form');
        form.appendChild(input);

        const formData = getFormData(form);

        expect(formData).toMatchSnapshot();
    })

    it('returns the correct value according to the snapshot, when the form does not have any inputs', () => {
        const form = document.createElement('form');

        const formData = getFormData(form);

        expect(formData).toMatchSnapshot();
    })
})
