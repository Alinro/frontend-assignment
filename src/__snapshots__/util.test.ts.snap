// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`test getFormData > returns the correct value according to the snapshot, when the form does not have any inputs 1`] = `"{}"`;

exports[`test getFormData > returns the correct value according to the snapshot, when the form has an input 1`] = `"{\\"inputName\\":\\"inputValue\\"}"`;
